src/adapters/RoomsAdapter.js

```diff
@@ -17,56 +33,55 @@ export const RoomType = {
  * This is a base class that defines the interface that maps room data.
  * Developers that want to extend `RoomsAdapter` must implement all of its methods,
  * adhering to the exact parameters and structure of the returned objects.
+ *
+ * `RoomsAdapter` handles data of a room such as details about that room but also
+ * activities that happened/will happen within that room. Activities are expected
+ * to be lazy loaded, and therefore, chunked.
  */
 export default class RoomsAdapter extends WebexAdapter {
-  /**
-   * A Room object with details about the room.
-   *
-   * @typedef {Object} Room
-   * @property {string}    ID      The room identifier.
-   * @property {string}    title   The room title.
-   * @property {RoomType}  type    The type of the room. @see RoomType enum
-   */
-
-  /**
-   * @typedef {Object} ActivityData
-   * @param {string} date The date to render a time ruler for. Must be a valid date-time string
-   *     https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#Date_Time_String_Format
-   */
-
   /**
    * Returns an observable that emits room data of the given ID.
    *
    * @param {string} ID  ID of the room to get.
    * @returns {Observable.<Room>}
    * @memberof RoomsAdapter
    */
-  // eslint-disable-next-line no-unused-vars
   getRoom(ID) {
     return throwError(new Error('getRoom(ID) must be defined in RoomsAdapter'));
   }
 
   /**
-   * Returns an observable that emits an array of previous activity IDs of the given roomID.
+   * Returns an observable that emits an array of current and future activities of the given roomID.
    *
-   * @param {string} ID  ID of the room to get.
-   * @returns {Observable.<Array.<string>>}
+   * @param {string} ID  ID of the room for which to get activities.
+   * @returns {Observable.<Array.<string|ActivityDate>>}
+   * @memberof RoomsAdapter
+   */
+  getRoomActivities(ID) {
+    return throwError(new Error('getRoomActivities(ID) must be defined in RoomsAdapter'));
+  }
+
+  /**
+   * Returns an observable that emits an array of the next chunk of previous
+   * activity data of the given roomID. If `hasMoreActivities` returns false,
+   * the observable will complete.
+   *
+   * @param {string} ID  ID of the room for which to get activities.
+   * @returns {Observable.<Array.<string|ActivityDate>>}
    * @memberof RoomsAdapter
    */
-  // eslint-disable-next-line no-unused-vars
   getPreviousRoomActivities(ID) {
     return throwError(new Error('getPreviousRoomActivities(ID) must be defined in RoomsAdapter'));
   }
 
   /**
-   * Returns an observable that emits an array of current activities of the given roomID.
+   * Returns `true` if there are more activities to load from the room of the given ID.
+   * Otherwise, it returns `false`.
    *
-   * @param {string} ID  ID of the room to get.
-   * @returns {Observable.<Array.<string|ActivityData>>}
+   * @param {string} ID ID of the room for which to verify activities.
```

Comment:
Missing the `@returns` statement

Reply:
There was an eslint error because the method doesn't actually return anything. I just disabled the warning for the file and added the return statement.